lib/matplotlib/backend_tools.py

```diff
@@ -318,7 +321,12 @@ class ToolQuit(ToolBase):
     default_keymap = rcParams['keymap.quit']
 
     def trigger(self, sender, event, data=None):
-        Gcf.destroy_fig(self.figure)
+        try:
+            manager = self.figure.canvas.manager
+        except:
```

Comment:
In two minds about this... on the one hand we just get an attribute/property and if we can't do that then we don't want to cleanup.  Because of that I put the more error prone part in the ``else`` block of the ``try``, this way we capture all errors relating just to accessing a property, note the following won't give an AttributeError
```python
class Foo(object):
  def __init__(self):
    self.p = {}

  @property
  def a(self):
    return self.p['a']

foo = Foo()
print(foo.a)  # raises KeyError
```

On the other hand, I agree with you and that in the above code we should replace the code with, but it just feels a bit like overkill...:
```python
@property
def a(self):
  try:
    ret = self.p['a']
  except KeyError:
    raise AttributeError